src/iidm/TieLine.cpp

```diff
@@ -16,119 +16,10 @@ namespace powsybl {
 
 namespace iidm {
 
-TieLine::HalfLine::HalfLine() :
-    m_lineCharacteristics(*this) {
-}
-
-double TieLine::HalfLine::getB1() const {
-    return m_lineCharacteristics.getB1();
-}
-
-double TieLine::HalfLine::getB2() const {
-    return m_lineCharacteristics.getB2();
-}
-
-double TieLine::HalfLine::getG1() const {
-    return m_lineCharacteristics.getG1();
-}
-
-double TieLine::HalfLine::getG2() const {
-    return m_lineCharacteristics.getG2();
-}
-
-const std::string& TieLine::HalfLine::getId() const {
-    return m_id;
-}
-
-std::string TieLine::HalfLine::getMessageHeader() const {
-    return stdcxx::format("Half line '%1%': ", m_id);
-}
-
-const std::string& TieLine::HalfLine::getName() const {
-    return m_name.empty() ? m_id : m_name;
-}
-
-double TieLine::HalfLine::getR() const {
-    return m_lineCharacteristics.getR();
-}
-
-double TieLine::HalfLine::getX() const {
-    return m_lineCharacteristics.getX();
-}
-
-double TieLine::HalfLine::getXnodeP() const {
-    return m_xnodeP;
-}
-
-double TieLine::HalfLine::getXnodeQ() const {
-    return m_xnodeQ;
-}
-
-bool TieLine::HalfLine::isFictitious() const {
-    return m_fictitious;
-}
-
-TieLine::HalfLine& TieLine::HalfLine::setB1(double b1) {
-    m_lineCharacteristics.setB1(b1);
-    return *this;
-}
-
-TieLine::HalfLine& TieLine::HalfLine::setB2(double b2) {
-    m_lineCharacteristics.setB2(b2);
-    return *this;
-}
-
-TieLine::HalfLine& TieLine::HalfLine::setFictitious(bool fictitious) {
-    m_fictitious = fictitious;
-    return *this;
-}
-
-TieLine::HalfLine& TieLine::HalfLine::setG1(double g1) {
-    m_lineCharacteristics.setG1(g1);
-    return *this;
-}
-
-TieLine::HalfLine& TieLine::HalfLine::setG2(double g2) {
-    m_lineCharacteristics.setG2(g2);
-    return *this;
-}
-
-void TieLine::HalfLine::setId(const std::string& id) {
-    m_id = id;
-}
-
-void TieLine::HalfLine::setName(const std::string& name) {
-    m_name = name;
-}
-
-void TieLine::HalfLine::setParent(TieLine& parent) {
-    m_parent = stdcxx::ref(parent);
-}
-
-TieLine::HalfLine& TieLine::HalfLine::setR(double r) {
-    m_lineCharacteristics.setR(r);
-    return *this;
-}
-
-TieLine::HalfLine& TieLine::HalfLine::setX(double x) {
-    m_lineCharacteristics.setX(x);
-    return *this;
-}
-
-TieLine::HalfLine& TieLine::HalfLine::setXnodeP(double xnodeP) {
-    m_xnodeP = xnodeP;
-    return *this;
-}
-
-TieLine::HalfLine& TieLine::HalfLine::setXnodeQ(double xnodeQ) {
-    m_xnodeQ = xnodeQ;
-    return *this;
-}
-
-TieLine::TieLine(const std::string& id, const std::string& name, bool fictitious, const std::string& ucteXnodeCode, HalfLine& half1, HalfLine& half2) :
+TieLine::TieLine(const std::string& id, const std::string& name, bool fictitious, const std::string& ucteXnodeCode, HalfLine&& half1, HalfLine&& half2) :
     Line(id, name, fictitious),
-    m_half1(attach(half1)),
-    m_half2(attach(half2)),
+    m_half1(std::move(attach(half1))),
+    m_half2(std::move(attach(half2))),
```

Comment:
To be consistent with the DanglingLine and ShuntCompensator, I would attach the TieLine to the half line after assign it

Reply:
Ok I call setParent on both half Line in constructor body.